File: Quivi.Backoffice.Api/Requests/ModifierGroups/CreateModifierGroupRequest.cs
using Quivi.Domain.Entities;

namespace Quivi.Backoffice.Api.Requests.ModifierGroups
{
    public class CreateModifierGroupRequest : ARequest
    {
        public required string Name { get; init; } = string.Empty;
        public int MinSelection { get; init; }
        public int MaxSelection { get; init; }
        public required IDictionary<string, CreateModifierItem> Items { get; init; } = new Dictionary<string, CreateModifierItem>();
        public IDictionary<Language, CreateModifierGroupLanguage>? Translations { get; init; }
    }

    public class CreateModifierItem
    {
        public decimal Price { get; init; }
        public int SortIndex { get; init; }
    }

    public class CreateModifierGroupLanguage
    {
        public required string Name { get; init; }
    }
}

File: quivi.backoffice.react.webapp/src/hooks/api/Dtos/modifierGroups/CreateModifierGroupRequest.ts
import { Language } from "../Language";

export interface CreateModifierGroupRequest {
    readonly name: string;
    readonly minSelection: number;
    readonly maxSelection: number;
    readonly items: Record<string, CreateModifierItem>;
    readonly translations?: Record<Language, CreateModifierGroupTransaltion>;
}

interface CreateModifierItem {
    readonly price: number;
    readonly sortIndex: number;
}

interface CreateModifierGroupTransaltion {
    readonly name: string;
}